common/src/main/java/org/apache/celeborn/common/write/InFlightRequestTracker.java

```diff
@@ -99,15 +136,19 @@ public boolean limitMaxInFlight(String hostAndPushPort) throws IOException {
     int currentMaxReqsInFlight = pushStrategy.getCurrentMaxReqsInFlight(hostAndPushPort);
 
     Set<Integer> batchIdSet = getBatchIdSetByAddressPair(hostAndPushPort);
+    LongAdder batchBytesSize = getBatchBytesSizeByAddressPair(hostAndPushPort);
     long times = waitInflightTimeoutMs / delta;
     try {
       while (times > 0) {
         if (cleaned) {
           // MapEnd cleans up push state, which does not exceed the max requests in flight limit.
           return false;
         } else {
-          if (totalInflightReqs.sum() <= maxInFlightReqsTotal
-              && batchIdSet.size() <= currentMaxReqsInFlight) {
+          if ((totalInflightReqs.sum() <= maxInFlightReqsTotal
+                  && batchIdSet.size() <= currentMaxReqsInFlight)
+              || (maxInFlightBytesSizeEnabled
+                  && totalInflightBytes.sum() <= maxInFlightBytesSizeTotal
+                  && batchBytesSize.sum() <= maxInFlightBytesSizePerWorker)) {
```

Comment:
My earlier query was, should this be `||` ?
As in, either total inflight is high, or inflight to a specific worker is high (assuming per worker threshold is lower than total !)

```suggestion
              || (maxInFlightBytesSizeEnabled && (
                  totalInflightBytes.sum() <= maxInFlightBytesSizeTotal ||
                  batchBytesSize.sum() <= maxInFlightBytesSizePerWorker))) {
```

Reply:
Looks like the PR got merged before this was addressed.
If this is not a concern, that should be fine - else let us do a follow up.

Reply:
To avoid this comment being ignored, I will submit a PR first